From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Shelley Vohr <shelley.vohr@gmail.com>
Date: Wed, 8 Jan 2020 10:12:46 -0800
Subject: fix: remove unused llhttp variables

Node.js updated llhttps in https://github.com/nodejs/node/pull/30553,
which updated some of the llparse_blobs. To keep numbering consistency,
it left the newly unused ones, which causes compilation failures in Electron
as we do not allow unused variables.

Fixed in https://github.com/nodejs/llparse/pull/34; this patch can be removed
as soon as llparse v6.2.1 is rolled into Node.js

diff --git a/src/implementation/c/compilation.ts b/src/implementation/c/compilation.ts
index b84b3e2..0750b43 100644
--- a/src/implementation/c/compilation.ts
+++ b/src/implementation/c/compilation.ts
@@ -77,6 +77,9 @@ export class Compilation {
         align = ` ALIGN(${blob.alignment})`;
       }
 
+      if (blob.alignment) {
+        out.push('#ifdef __SSE4_2__');
+      }
       out.push(`static const unsigned char${align} ${blob.name}[] = {`);
 
       for (let i = 0; i < buffer.length; i += BLOB_GROUP_SIZE) {
@@ -103,6 +106,9 @@ export class Compilation {
       }
 
       out.push(`};`);
+      if (blob.alignment) {
+        out.push('#endif  /* __SSE4_2__ */');
+      }
     }
     out.push('');
   }
